components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/JsonSchemaValidator.java

```diff
@@ -0,0 +1,188 @@
+     *                       API which was enabled the JSON_Validator message mediation in flow.
+     * @return a boolean true if the message content is passed the json schema criteria.
+     */
+    public boolean mediate(MessageContext messageContext) {
+        if (log.isDebugEnabled()) {
+            log.debug("JSON schema validation mediator is activated...");
+        }
+        Map<String, InputStream> inputStreams = null;
+        org.apache.axis2.context.MessageContext axis2MC;
+        Boolean validRequest = true;
+        String contentType;
+        String apiContext;
+        String requestMethod;
+        axis2MC = ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        contentType = axis2MC.getProperty(ThreatProtectorConstants.CONTENT_TYPE).toString();
+        apiContext = messageContext.getProperty(ThreatProtectorConstants.API_CONTEXT).toString();
+        requestMethod = axis2MC.getProperty(ThreatProtectorConstants.HTTP_REQUEST_METHOD).toString();
+
+        if (!APIConstants.SupportedHTTPVerbs.GET.name().equalsIgnoreCase(requestMethod) &&
+                (ThreatProtectorConstants.APPLICATION_JSON.equals(contentType) ||
+                        ThreatProtectorConstants.TEXT_JSON.equals(contentType))) {
+            JSONConfig jsonConfig = configureSchemaProperties(messageContext);
+            ConfigurationHolder.addJsonConfig(jsonConfig);
+            APIMThreatAnalyzer apimThreatAnalyzer = AnalyzerHolder.getAnalyzer(contentType);
+            try {
+                inputStreams = GatewayUtils.cloneRequestMessage(messageContext);
+                if (inputStreams != null) {
+                    InputStream inputStreamJson = inputStreams.get(ThreatProtectorConstants.JSON);
+                    BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStreamJson);
+                    apimThreatAnalyzer.analyze(bufferedInputStream, apiContext);
+                }
+            } catch (APIMThreatAnalyzerException e) {
+                validRequest = false;
+                String message = "Request is failed due to JSON schema validation failure: ";
+                GatewayUtils.handleThreat(messageContext, ThreatProtectorConstants.HTTP_SC_CODE,
+                        message + e.getMessage());
+            } catch (IOException e) {
+                String message = "Error occurred while building the request: ";
+                GatewayUtils.handleThreat(messageContext, ThreatProtectorConstants.HTTP_SC_CODE,
+                        message + e.getMessage());
+            } finally {
+                // return analyzer to the pool
+                AnalyzerHolder.returnObject(apimThreatAnalyzer);
+            }
+        } else {
+            GatewayUtils.handleThreat(messageContext, APIMgtGatewayConstants.HTTP_SC_CODE,
+                    APIMgtGatewayConstants.REQUEST_TYPE_FAIL_MSG);
+        }
+        GatewayUtils.setOriginalInputStream(inputStreams, axis2MC);
+        if (validRequest) try {
```

Comment:
if condition with "{". According to WSO2 convention, all the if conditions should be having {} for readability